### DISCUSSION TITLE
[Feature request] 期望在koa模块中的Application或Framework实现一个函数，传入url, headers, data等参数能够模拟成一个真正的请求调用命中的方法，触发中间件、拦截器等midway能力

### DISCUSSION THREAD

**Discussion by os023** [midwayjs/midway#4175] [Ideas]
### The feature, motivation（功能、动机）

该功能可用于在cronJob, Bull等辅助模块中调用koa，复用既有中间件、拦截器中的数据构造、错误处理、日志记录等能力

### Alternatives（替代方案）

httpService

**Comment by waitingsong** [comment]
这个不行么？ https://midwayjs.org/docs/mock

**Reply by os023** [reply]
这个的本质还是httpService

**Comment by czy88840616** [comment]
这个实现不了

 - 1、走中间件意味着需要去模拟整个完整的 http ctx，这个在非 http 场景是不太可能的
 - 2、bull -> koa 属于内部自己调用，属于完全的可信调用，没必要再绕一次中间件（鉴权等），如果需要拿到用户信息，应该使用另外手段(比如 token 等）

**Reply by os023** [reply]
我有一个思路，通过routeService获取匹配url的class和method，构造一个新的context，将data, headers, 等信息附加到这个context上，这样执行命中的method时让他将这个context视为正常请求的ctx，现在我不知道该怎么构造这个context并把context传给method
